server/src/Application/Repositories/IRepository.cs

```diff
@@ -4,9 +4,9 @@ namespace StudyZen.Application.Repositories;
 
 public interface IRepository<TEntity> where TEntity : BaseEntity
 {
-    void Add(TEntity instance);
-    TEntity? GetById(int instanceId);
-    List<TEntity> GetAll();
-    bool Update(TEntity instance);
-    bool Delete(int instanceId);
+    Task Add(TEntity instance);
```

Comment:
It's a standard practice to add `Async` suffix to methods which return a task:
https://stackoverflow.com/questions/15951774/does-the-use-of-the-async-suffix-in-a-method-name-depend-on-whether-the-async